tests/integration/suite/test_users.py

```diff
@@ -16,3 +18,62 @@ def test_user_cant_deactivate_self(admin_mc):
         client.update(admin_mc.user, enabled=False)
 
     assert e.value.error.status == 422
+
+
+def test_globalrolebinding_create(admin_mc, remove_resource):
+    client = admin_mc.client
+    grb = client.create_globalRoleBinding(globalRoleId="admin",
+                                          userId="u-" + random_str())
+    remove_resource(grb)
+    assert grb.annotations["field.cattle.io/grbUpgrade"] == "true"
+
+    # create a grb without the rancher api with a bad finalizer
+    api = CustomObjectsApi(admin_mc.k8s_client)
+    json = {
+        "apiVersion": "management.cattle.io/v3",
+        "globalRoleName": "admin",
+        "kind": "GlobalRoleBinding",
+        "metadata": {
+            "finalizers": [
+                "clusterscoped.controller.cattle.io/grb-sync_fake",
+            ],
+            "generation": 1,
+            "name": "grb-" + random_str(),
+        },
+        "userName": "u-" + random_str()
+    }
+    grb_k8s = api.create_cluster_custom_object(group="management.cattle.io",
+                                               version="v3",
+                                               plural="globalrolebindings",
+                                               body=json
+                                               )
+    remove_resource(grb_k8s["metadata"]["name"])
+    grb_k8s = api.get_cluster_custom_object(group="management.cattle.io",
```

Comment:
After it's created you should be able to use the normal rancher client to interact with it going forward, that would help cleanup the code

Reply:
I use the rancher client more in the code now, I still use the k8s client for access to the finalizers though